src/app/login/login.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { UserAuthenticationService } from '../services/user-authentication.service';
+import { first } from 'rxjs/operators';
+import { AlertService } from '../services/alert.service';
+
+@Component({
+    selector: 'app-login',
+    templateUrl: './login.component.html',
+    styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+    loginForm: FormGroup;
+    loading = false;
+    submitted = false;
+    returnUrl: string;
+    error = '';
+
+    constructor(
+        private formBuilder: FormBuilder,
+        private route: ActivatedRoute,
+        private router: Router,
+        private userAuthenticationService: UserAuthenticationService,
+        private alertService: AlertService
+    ) {}
+
+    ngOnInit() {
+        this.createForm();
+
```

Comment:
Why have you started to put spaces between lines? In this case, no need for that